client/my-sites/earn/memberships/test/coupons-list.jsx

```diff
@@ -0,0 +1,337 @@
+		const render = ( el, options ) =>
+			renderWithProvider( el, {
+				...options,
+				initialState: {
+					...initialState,
+					memberships: {
+						settings: {
+							1: {
+								couponsAndGiftsEnabled: true,
+								connectedAccountId: 123,
+								connectedAccountMinimumCurrency: {
+									USD: 0.5,
+								},
+							},
+						},
+						couponList: {
+							items: {
+								1: [ testCoupon1, testCoupon2, testCoupon3 ],
+							},
+						},
+						productList: {
+							items: {
+								1: [ testProduct1, testProduct2 ],
+							},
+						},
+					},
+				},
+				reducers: {
+					ui: uiReducer,
+					memberships: membershipsReducer,
+					siteSettings: siteSettingsReducer,
+				},
+			} );
+		render( <CouponsList /> );
+		const addCouponButton = getAddCouponButton();
+		const missingAddEditCouponDialog = queryAddEditCouponDialog();
+		const missingDiscountValueInput = queryDiscountValueInput();
+		const missingDiscountPercentageInput = queryDiscountPercentageInput();
+
+		// TODO: improve a11y for coupon list to be able to access toggle menu and assert that each coupon has been rendered without snapshot
+
+		expect( addCouponButton ).toBeInTheDocument();
+		expect( missingAddEditCouponDialog ).not.toBeInTheDocument();
+		expect( missingDiscountPercentageInput ).not.toBeInTheDocument();
+		expect( missingDiscountValueInput ).not.toBeInTheDocument();
+	} );
+
+	test( 'should display new coupon modal when clicking add coupon', async () => {
+		const user = userEvent.setup();
+		const render = ( el, options ) =>
+			renderWithProvider( el, {
+				...options,
+				initialState: {
+					...initialState,
+					memberships: {
+						settings: {
+							1: {
+								couponsAndGiftsEnabled: true,
+								connectedAccountId: 123,
+								connectedAccountMinimumCurrency: {
+									USD: 0.5,
+								},
+							},
+						},
+						couponList: {
+							items: {
+								1: [ testCoupon1, testCoupon2, testCoupon3 ],
+							},
+						},
+						productList: {
+							items: {
+								1: [ testProduct1, testProduct2 ],
+							},
+						},
+					},
+				},
+				reducers: {
+					ui: uiReducer,
+					memberships: membershipsReducer,
+					siteSettings: siteSettingsReducer,
+				},
+			} );
+		render( <CouponsList /> );
+
+		await user.click( screen.getByRole( 'button', { name: 'Add a new coupon' } ) );
+
+		const codeTextbox = getCodeTextbox();
+		const discountTypeSelect = getDiscountTypeSelect();
+		const discountPercentageInput = getDiscountPercentageInput();
+		const discountValueInput = queryDiscountValueInput();
+		const limitDurationToggle = getLimitDurationToggle();
+		const limitDurationSelection = getLimitDurationSelection();
+		const limitCouponsToggle = getLimitCouponsToggle();
+		const limitCouponsTextInput = getLimitCouponsTextInput();
+
+		expect( codeTextbox ).not.toHaveValue();
+		expect( discountTypeSelect ).toHaveValue( COUPON_DISCOUNT_TYPE_PERCENTAGE );
+		expect( discountPercentageInput ).toBeInTheDocument();
```

Comment:
When testing a visible element, it's always recommended to use `.toBeVisible()` instead of `.toBeInTheDocument()`. An element can be in the document tree, but be invisible after all, while we want it to be visible.